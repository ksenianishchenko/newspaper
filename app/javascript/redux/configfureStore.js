import { createStore, compose, applyMiddleware } from 'redux';
import {combineReducers} from "redux";
import thunk from 'redux-thunk';
import logger from "redux-logger";
import {userReducer} from "./user/user-reducer";
import {profileReducer} from "./profile/profile-reducer";
import {tweetReducer} from "./tweet/tweet-reducer";

export const rootReducer = combineReducers({
  user: userReducer,
  profile: profileReducer,
  tweet: tweetReducer
});

const middlewares = [logger, thunk];

let store;

if (window.__REDUX_DEVTOOLS_EXTENSION__) {
  store = createStore(
    rootReducer,
    compose(
      applyMiddleware(...middlewares),
      window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    )
  );
} else {
  store = createStore(
    rootReducer,
    compose(
      applyMiddleware(...middlewares)
    )
  );
}

export default store;
