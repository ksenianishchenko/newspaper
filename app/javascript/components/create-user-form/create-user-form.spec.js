import React from "react";
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import { rootReducer } from "../../redux/configfureStore";
import thunk from 'redux-thunk';

import CreateUserForm from "./create-user-form";

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('<CreateUserForm /> unit test', () => {
    let wrapper;
    const setState = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation((init) => [init, setState]);
    const middlewares = [thunk];
    const mockStore = createStore(
        rootReducer, compose(
        applyMiddleware(...middlewares)
    ));

    beforeEach(() => {
        wrapper = mount(<Provider store={mockStore}>
            <CreateUserForm />
        </Provider>
        );
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render form with inputs', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("input")).toHaveLength(4);
        expect(wrapper.find('button[type="submit"]').exists()).toEqual(true);
    });

    it('should capture email value onChange', () => {
      wrapper.find('input[name="email"]').simulate('change', {target: { value: 'the-value' }});
      expect(setState).toBeTruthy();
    });
  });

