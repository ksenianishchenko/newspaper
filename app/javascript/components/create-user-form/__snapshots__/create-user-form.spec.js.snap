// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUserForm /> unit test should render form with inputs 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(CreateUserForm)>
    <CreateUserForm
      handleSignUpRequest={[Function]}
    >
      <form
        action=""
        className="create-user-form p-3"
        method=""
      >
        <h2
          className="h4 mb-4"
        >
          Add new user
        </h2>
        <FormInput
          className="form-control mb-4"
          handleChange={[Function]}
          id="email"
          name="email"
          placeholder="email"
        >
          <input
            className="form-control mb-4"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="email"
          />
        </FormInput>
        <FormInput
          className="form-control mb-4"
          handleChange={[Function]}
          id="handle"
          name="handle"
          placeholder="user_name"
        >
          <input
            className="form-control mb-4"
            id="handle"
            name="handle"
            onChange={[Function]}
            placeholder="user_name"
          />
        </FormInput>
        <FormInput
          className="form-control mb-4"
          handleChange={[Function]}
          id="password"
          name="password"
          placeholder="password"
        >
          <input
            className="form-control mb-4"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="password"
          />
        </FormInput>
        <FormInput
          className="form-control mb-4"
          handleChange={[Function]}
          id="password"
          name="password"
          placeholder="password"
        >
          <input
            className="form-control mb-4"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="password"
          />
        </FormInput>
        <Button
          className="btn btn-primary btn-block my-4 waves-effect waves-light"
          onClick={[Function]}
          type="submit"
        >
          <button
            className="btn btn-primary btn-block my-4 waves-effect waves-light"
            onClick={[Function]}
            type="submit"
          >
             Submit 
          </button>
        </Button>
      </form>
    </CreateUserForm>
  </Connect(CreateUserForm)>
</Provider>
`;
